src/dune_engine/scheduler.ml

```diff
@@ -1080,6 +1084,18 @@ let poll (config : Config.t) ~once ~finally =
   | None -> Exn.raise exn
   | Some bt -> Exn.raise_with_backtrace exn bt
 
+let go config run =
+  let t =
+    prepare config ~polling:false
+      ~handler:
+        { new_event = (fun _ -> ())
```

Comment:
Don't we need `new_event` to still update the status line even when not polling?
(I'm sorry if I'm the one who introduced this bug, I'm not 100% sure of my refactoring)

Reply:
`new_event` needs to be reported when `--display=progress`, so you're right.